lua/compe/init.lua

```diff
@@ -1,84 +1,128 @@
-  self.completion:register_source(Source.new(id, source))
+--- registers a new lua source
+-- passes params using new func from `source` to `completion` register_source
+-- func.
+-- @param id string representing the source name.
+-- @param obj table representing nvim-compe source.
+-- @function register_lua_source
+compe.register_lua_source = function(self, id, obj)
+  self.completion:register_source(source.new(id, obj))
 end
 
---- register_vim_source
-function Compe.register_vim_source(self, id)
-  self.completion:register_source(Source.new(id, VimBridge.new(id)))
+--- registers a new vim source
+-- passes params using new func from `source` and `vim_bridge` to `completion`
+-- register_source func.
+-- @param id string representing the source name.
+-- @function register_vim_source
+compe.register_vim_source = function(self, id)
+  self.completion:register_source(source.new(id, vim_bridge.new(id)))
 end
 
---- unregister_source
-function Compe.unregister_source(self, id)
+--- unregisters a vim or lua source
+-- passes params using to `completion` unregister_source func.
+-- @param id string representing the source name.
+-- @function unregister_source
+compe.unregister_source = function(self, id)
   self.completion:unregister_source(id)
 end
 
---- on_complete_changed
-function Compe.on_complete_changed(self)
+--- wraps `completion:on_complete_changed`
+-- ???
+-- @param self
+-- @function on_complete_changed
+compe.on_complete_changed = function(self)
   local status, value = pcall(function() self.completion:on_complete_changed() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_complete_done
-function Compe.on_complete_done(self)
+--- wraps `completion:on_complete_done`
+-- ???
+-- @param self
+-- @function on_complete_done
+compe.on_complete_done = function(self)
   local status, value = pcall(function() self.completion:on_complete_done() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_text_changed
-function Compe.on_text_changed(self)
+--- wraps `completion:on_text_changed`
+-- Event that triggers on text change ???
+-- @param self
+-- @function on_text_changed
+compe.on_text_changed = function(self)
   local status, value = pcall(function() self.completion:on_text_changed() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_insert_leave
-function Compe.on_insert_leave(self)
+--- wraps `completion:on_text_changed`
+-- Event that triggers on insert leave ???
+-- @param self
+-- @function on_insert_leave
+compe.on_insert_leave = function(self)
   local status, value = pcall(function() self.completion:on_insert_leave() end)
   if not(status) then
-    Debug:log(value)
+    debug:log(value)
   end
 end
 
---- on_manual_complete
-function Compe.on_manual_complete(self)
+--- wraps `completion:on_manual_complete`
+-- ???
+-- @param self
+-- @function on_manual_complete
+compe.on_manual_complete = function(self)
```

Comment:
I'm thinking now, It should be `manual_complete`.

It will use to trigger completion manually.